Sources/SourceKitLSP/SourceKitServer.swift

```diff
@@ -58,6 +58,84 @@ enum LanguageServerType: Hashable {
   }
 }
 
+/// Keeps track of the state to send work done progress updates to the client
+class WorkDoneProgressState {
```

Comment:
nit: I see no subclasses of this class in this PR and probably none are anticipated in future changes, so this could be `final`?
```suggestion
final class WorkDoneProgressState {
```